5.5/run-mysqld.sh

```diff
@@ -34,8 +34,14 @@ EOSQL
 
 # The MYSQL_ROOT_PASSWORD is optional
 if [ -v MYSQL_ROOT_PASSWORD ]; then
-mysql $mysql_flags <<EOSQL
-    SET PASSWORD FOR 'root'@'%' = PASSWORD('${MYSQL_ROOT_PASSWORD}');
+  mysql $mysql_flags <<EOSQL
+    GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' IDENTIFIED BY '${MYSQL_ROOT_PASSWORD}';
+EOSQL
+else
+  mysql $mysql_flags <<EOSQL
```

Comment:
Can you add a comment here explaining why you're doing a gran and a drop right after it?